pkg/kv/postgres/store.go

```diff
@@ -319,46 +323,18 @@ func (s *Store) Scan(ctx context.Context, partitionKey []byte, options kv.ScanOp
 	if options.BatchSize != 0 && s.Params.ScanPageSize != 0 && options.BatchSize < s.Params.ScanPageSize {
 		limit = options.BatchSize
 	}
-
-	entries, err := s.scanInternal(ctx, partitionKey, options.KeyStart, limit, true)
-	if err != nil {
-		return nil, err
-	}
-	return &EntriesIterator{
+	it := &EntriesIterator{
 		ctx:          ctx,
+		partitionKey: partitionKey,
+		startKey:     options.KeyStart,
 		store:        s,
-		entries:      entries,
 		limit:        limit,
-		currEntryIdx: -1,
-	}, nil
-}
-
-func (s *Store) scanInternal(ctx context.Context, partitionKey []byte, keyStart []byte, limit int, includeStart bool) ([]kv.Entry, error) {
-	var (
-		rows pgx.Rows
-		err  error
-	)
-
-	if keyStart == nil {
-		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 ORDER BY key LIMIT $2`, partitionKey, limit)
-	} else {
-		compareOp := ">="
-		if !includeStart {
-			compareOp = ">"
-		}
-		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 AND key `+compareOp+` $2 ORDER BY key LIMIT $3`, partitionKey, keyStart, limit)
 	}
-	if err != nil {
-		return nil, fmt.Errorf("postgres scan: %w", err)
-	}
-	defer rows.Close()
-
-	var entries []kv.Entry
-	err = pgxscan.ScanAll(&entries, rows)
-	if err != nil {
-		return nil, fmt.Errorf("scanning all entries: %w", err)
+	it.runQuery(true)
+	if it.err != nil {
+		return nil, it.err
```

Comment:
Close iterator?

Reply:
No real meaning to closing this iterator - see other comment.